src/nft/components/sell/select/SelectPage.tsx

```diff
@@ -1,3 +1,380 @@
+import InfiniteScroll from 'react-infinite-scroll-component'
+import { useInfiniteQuery, useQuery } from 'react-query'
+
+import * as styles from './SelectPage.css'
+
+const formatEth = (price: number) => {
+  if (price > 1000000) {
+    return `${Math.round(price / 1000000)}M`
+  } else if (price > 1000) {
+    return `${Math.round(price / 1000)}K`
+  } else {
+    return `${Math.round(price * 100 + Number.EPSILON) / 100}`
+  }
+}
+
 export const SelectPage = () => {
-  return <div>Select NFTs to list</div>
+  const { address } = useWalletBalance()
+  const collectionFilters = useWalletCollections((state) => state.collectionFilters)
+
+  const { data: ownerCollections } = useQuery(
+    ['ownerCollections', address],
+    () => OSCollectionsFetcher({ params: { asset_owner: address, offset: '0', limit: '300' } }),
+    {
+      refetchOnWindowFocus: false,
+    }
+  )
+
+  const ownerCollectionsAddresses = useMemo(() => ownerCollections?.map(({ address }) => address), [ownerCollections])
+  const { data: collectionStats } = useQuery(
+    ['ownerCollectionStats', ownerCollectionsAddresses],
+    () => fetchMultipleCollectionStats({ addresses: ownerCollectionsAddresses ?? [] }),
+    {
+      refetchOnWindowFocus: false,
+    }
+  )
+
+  const {
+    data: ownerAssetsData,
+    fetchNextPage,
+    hasNextPage,
+    isSuccess,
+  } = useInfiniteQuery(
+    ['ownerAssets', address, collectionFilters],
+    async ({ pageParam = 0 }) => {
+      return await fetchWalletAssets({
+        ownerAddress: address ?? '',
+        collectionAddresses: collectionFilters,
+        pageParam,
+      })
+    },
+    {
+      getNextPageParam: (lastPage, pages) => {
+        return lastPage?.flat().length === 25 ? pages.length : null
+      },
+      refetchOnWindowFocus: false,
+      refetchOnMount: false,
+    }
+  )
+
+  const ownerAssets = useMemo(() => (isSuccess ? ownerAssetsData?.pages.flat() : null), [isSuccess, ownerAssetsData])
+
+  const walletAssets = useWalletCollections((state) => state.walletAssets)
+  const setWalletAssets = useWalletCollections((state) => state.setWalletAssets)
+  const displayAssets = useWalletCollections((state) => state.displayAssets)
+  const setDisplayAssets = useWalletCollections((state) => state.setDisplayAssets)
+  const setWalletCollections = useWalletCollections((state) => state.setWalletCollections)
+  const sellAssets = useSellAsset((state) => state.sellAssets)
+  const reset = useSellAsset((state) => state.reset)
+  const setSellPageState = useSellPageState((state) => state.setSellPageState)
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    setWalletAssets(ownerAssets?.flat() ?? [])
+  }, [ownerAssets, setWalletAssets])
+
+  useEffect(() => {
+    ownerCollections && setWalletCollections(ownerCollections)
+  }, [ownerCollections, setWalletCollections])
+
+  const listFilter = useWalletCollections((state) => state.listFilter)
+
+  useEffect(() => {
```

Comment:
slightly confused about the need for two separate hooks here. this and L103

Reply:
Good catch, the hook at 103 is extraneous and can be removed